src/Apps/Order2/Routes/Checkout/CheckoutContext/Order2CheckoutContext.tsx

```diff
@@ -243,6 +246,16 @@ const useBuildCheckoutContext = (
     [currentStepName],
   )
 
+  const setConfirmationToken = useCallback(
+    ({ confirmationToken }: { confirmationToken: any }) => {
+      dispatch({
+        type: "PAYMENT_COMPLETE",
+        payload: { confirmationToken: confirmationToken },
+      })
+    },
+    [state.confirmationToken], // Is this correct?
```

Comment:
| `// Is this correct?`

I don't think so - `useCallback()` is just a way to memoize a function so it isn't re-initialized on every render in a functional component. similar to `useEffect` and `useMemo` the second argument dependency array is  a list of things to check to know whether something has gotten stale and the function needs to be re-initialized. Since `state.confirmationToken` isn't referenced inside the function body you don't need it.

As a counter example say you had the line `if (confirmationToken === state.confirmationToken) return` - that would mean it has to go in the dependency array. Generally the linter will tell you what dependencies are missing and even offer to add them for you.